pkg/importer/vddk-datasource.go

```diff
@@ -668,6 +680,37 @@ func (sink *VDDKFileSink) Write(buf []byte) (int, error) {
 	return written, err
 }
 
+// ZeroRange fills the destination range with zero bytes
+func (sink *VDDKFileSink) ZeroRange(offset uint64, length uint32) error {
+	var err error
+	if !sink.isBlock { // For files, see if it's possible to punch a hole
+		err := syscall.Fallocate(int(sink.file.Fd()), unix.FALLOC_FL_ZERO_RANGE, int64(offset), int64(length))
+		if err != nil {
+			klog.Errorf("Unable to zero range on destination, falling back to pwrite: %v", err)
+		}
+	} // For block devices, just use pwrite for now. Replace with BLKZEROOUT later if possible.
+
+	if err != nil { // Fall back to regular pwrite
```

Comment:
I'm surprised that this code block doesn't write in the block case.
Is that intentional?

Reply:
It was intentional to avoid the fallocate on block devices, but also wrong - it looks like ZERO_RANGE is actually supported on block devices after all, so I will update it to do that.

The original intention was: for files, try ZERO_RANGE and fall back to pwrite if it fails; for block devices, always skip straight to pwrite. The pwrite fallback is supposed to reset err, so I will also make sure it does that.